rasa-ext-files/src/Rasa/Ext/Files/Internal/Events.hs

```diff
@@ -0,0 +1,18 @@
+module Rasa.Ext.Files.Internal.Events where
+
+import Rasa.Ext
+
+import Control.Monad
+
+data Extension = Extension String | UnknownExtension deriving (Eq)
+
+data FileLoaded
+  = FileLoaded Extension BufRef
```

Comment:
I think the `FileLoaded` event should probably just pass the whole filepath and the consumer can pull the extension out themselves if they like.